xdsl/transforms/canonicalization_patterns/csl.py

```diff
@@ -81,21 +93,27 @@ def match_and_rewrite(self, op: csl.GetMemDsdOp, rewriter: PatternRewriter) -> N
             stride_op := next(iter(op.result.uses)).operation, csl.SetDsdStrideOp
         ):
             return
-        # only works on 1d
-        if op.offsets and len(op.offsets) > 1:
+        # only works on 1d and default (unspecified) tensor_access
+        if len(op.sizes) > 1 or op.tensor_access:
             return
 
         # check if we can promote arith.const to property
         if (
             isinstance(stride_op.stride, OpResult)
             and isinstance(cnst := stride_op.stride.op, arith.ConstantOp)
-            and isattr(cnst.value, AnyIntegerAttrConstr)
+            and isa(attr_val := cnst.value, AnyIntegerAttr)
         ):
+            tensor_access = AffineMap.from_callable(
+                lambda x: (x * attr_val.value.data,)
+            )
             rewriter.replace_matched_op(
```

Comment:
Is there a way to not need to suppress all the warnings?

If not, could you please comment why they're suppressed?